packages/sitecore-jss-react/src/components/Text.tsx

```diff
@@ -90,10 +90,16 @@ export const Text: FunctionComponent<TextProps> = ({
 };
 
 Text.propTypes = {
-  field: PropTypes.shape({
-    value: PropTypes.string,
-    editable: PropTypes.string,
-  }),
+  field: PropTypes.oneOfType([
```

Comment:
Instead of two separate shapes, you can write:
```
field: PropTypes.shape({
   value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
   editable: PropTypes.string
})
```

Reply:
Fixed